server/api/crud/runs.py

```diff
@@ -468,3 +453,48 @@ async def _post_delete_run(project, uid):
                 project,
                 uid,
             )
+
+    @staticmethod
+    def _get_artifacts_from_uris(db_session, project, run):
```

Comment:
I'm not sure, but should we use TTLcache for this function?
IIUC, in case we have lots of artifacts and there are multiple users that try to get them, we will cause heavy load on the db.

Reply:
Interesting.. though the artifacts are per run so i don't think caching will help here since we don't expect many requests for the same run.